pkg/telemetry/phonehome/segment/segment.go

```diff
@@ -0,0 +1,121 @@
+package segment
+
+import (
+	"time"
+
+	segment "github.com/segmentio/analytics-go"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	log = logging.LoggerForModule()
+)
+
+// Enabled tells whether telemetry data collection is enabled.
+func Enabled() bool {
+	return env.TelemetryStorageKey.Setting() != ""
+}
+
+type segmentTelemeter struct {
+	client   segment.Client
+	userID   string
+	identity map[string]any
+}
+
+func (t *segmentTelemeter) Identify(props map[string]any) {
+	traits := segment.NewTraits()
+	identity := segment.Identify{
+		UserId: t.userID,
+		Traits: traits,
+	}
+
+	if t.identity != nil {
+		for k, v := range t.identity {
+			traits.Set(k, v)
+		}
+	}
+	for k, v := range props {
+		traits.Set(k, v)
+	}
+	log.Info("Identifying with ", identity)
+	if err := t.client.Enqueue(identity); err != nil {
+		log.Error("Cannot enqueue Segment identity event: ", err)
+	}
+}
+
+// NewTelemeter creates and initializes a Segment telemeter instance.
+func NewTelemeter(userID string, identity map[string]any) *segmentTelemeter {
+	key := env.TelemetryStorageKey.Setting()
+	server := ""
+	return initSegment(userID, identity, key, server)
+}
+
+type logWrapper struct {
+	internal *logging.Logger
+}
+
+func (l *logWrapper) Logf(format string, args ...any) {
+	l.internal.Logf(zapcore.InfoLevel, format, args...)
+}
+
+func (l *logWrapper) Errorf(format string, args ...any) {
+	l.internal.Errorf(format, args...)
+}
+
+func initSegment(userID string, identity map[string]any, key, server string) *segmentTelemeter {
+	segmentConfig := segment.Config{
+		Endpoint: server,
+		Interval: 5 * time.Minute,
```

Comment:
I saw the environment variable `ROX_TELEMETRY_FREQUENCY`. Should this be reused here, or is it from the "old" telemetry implementation (so we can just remove it to avoid confusion)?

Reply:
The variable is from the legacy/parallel implementation, and it's not used.
Nice idea, it might be taken for this use I think.